### ISSUE TITLE
Parameter Estimation for Linear Regression for IHEPC dataset 

### ISSUE THREAD

**Issue by MrPr3ntice** [White-Link/UnsupervisedScalableRepresentationLearningTimeSeries#15]
Is there a reason for not using _deterministic_ OLS-Estimation (via Moore-Penrose-Pseudoinverse, e.g. in _sklearn.linear_model.LinearRegression.fit()_) in case of the linear regression for the IHEPC dataset? Especially for reproducibility purposes, this may be benefitial (even if it is assumed here that the errors are normally distributed, so that the OLS-estimator also corresponds to the Maximum-Likelihood-optimal-estimator). Maybe OLS is unfavorable because of hypothetical high matrix dimensions and therefore the here used sequential adam optimizer is computationally superior?

Thanks in advance!

**Comment by White-Link** [comment]
Hi!

Indeed, the reason behind this choice is practical. We wanted to keep the same optimizer for both training settings (with raw time series values and with representations), while keeping a reasonable training time for the predictor. Using a gradient descent algorithm should be computationally less expensive for approximately the same performances on such a large dataset and for a convex objective.

Besides, the purpose of this task is merely to show the applicability of our method to long time series and to other tasks than classifications, hence the simplicity of the approach. I am sure there must be smarter ways to tackle it, especially by designing data-specific methods, which we did not perform here.

**Comment by MrPr3ntice** [comment]
Hi,
thanks for your answer, that sounds plausible!
